Frontend/student-hub/src/pages/MentorshipPage.js

```diff
@@ -264,93 +288,164 @@ const delayTime = 60000;
   }
 
   return (
-    <Container maxWidth="sm">
-      <Grid container spacing={2} alignItems="center" justify="center" style={{ minHeight: '80vh' }}>
+    <Container maxWidth="md">
+      <Grid container spacing={3}>
+        <Grid item xs={12}>
+        </Grid>
+
         <Grid item xs={12}>
-        <SearchContainer>
-        <SearchInput
-          placeholder="Search Boards"
-          inputProps={{ 'aria-label': 'search boards' }}
-          value={searchTerm}
-          onChange={handleSearch}
-        />
-        <SearchIconButton type="submit"  aria-label="search">
-          <SearchIcon />
-        </SearchIconButton>
-      </SearchContainer>
-
-{userRole === 'Mentor' ? (
-  <Grid container spacing={2} alignItems="center" justify="center">
-    <Grid item xs={12}>
-      <Button onClick={handleOpenMenteeList}>Mentee List</Button>
-      <Button onClick={handlePendingRequest}>Pending Request</Button>
-    </Grid>
-  </Grid>
-) : (
-  <Grid container spacing={2} alignItems="center" justify="center">
-    <Grid item xs={12}>
-      <Button onClick={handleOpenMentorList}>Mentor</Button>
-    </Grid>
-  </Grid>
-)}
+          <SearchContainer>
+            <SearchInput
+              placeholder="Search Mentorship"
+              value={searchTerm}
+              onChange={handleSearch}
+            />
+            <SearchIconButton>
+              <SearchIcon />
+            </SearchIconButton>
+          </SearchContainer>
         </Grid>
-       {filteredMentorship.map((mentorship) => {
-        if (mentorship.mentorship === 'Mentor') {
-          return (
-        <Grid item xs={12} sm={6} md={4} key={mentorship.id}>
-          <Tooltip
+        <Grid item xs={12}>
+          {userRole !== 'Mentor' && (
```

Comment:
it would probably be cleaner if you move these into separate components for MentorList and MenteeList that way the conditional logic in the parent Page component isn't so complex

Reply:
also feels like we could have a reusable component for all of the CardContent since that UI is being reused across Mentor and Mentee Lists